### PR TITLE
Issue/612/teams tracks on map for admin

### LINKED ISSUE #612: Teams tracks on map for admin

### PR DESCRIPTION
**Pull request by Bedpaw**
- [ ] Fix teamColor issue and show field in join event
- [ ] Admin can see all teams track reflected with color
- [ ] User can see all points, his points has stroke color with teamColor
- [ ] Changes in downloadEvent for admin to fetch teams data

close #612

### CODE REVIEW

File: client/src/components/pages/join-event.vue [hunk 1]
```diff
@@ -70,18 +80,20 @@ export default {
...
     const teamName = ref('');
+    const teamColor = ref('');
     const nickname = ref('');
     const event = ref(null);
     const eventConfirmationPopUp = ref(null);
     const form = useForm();
     const store = useStore();
 
-    const shouldDisplayTeamNameInput = computed(() => event?.value?.role === ACCOUNT_TYPES.teamLeader);
+    const shouldDisplayTeamDetails = computed(() => event?.value?.role === ACCOUNT_TYPES.teamLeader);
```
Comment (dbetka) [inline client/src/components/pages/join-event.vue:96]: it's not necessary to use question mark after event - it's always defined - only value of event can by null.

File: client/src/store/team.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
 import { Module } from 'vuex';
 import { TeamDTO } from 'models/dtos/team';
 import { PointType } from 'models/point';
+import { api } from 'api';
+import { Team } from 'models/team';
 
 const initState = () => ({
   teamId: '',
   teamName: '',
+  teamColor: '#447744',
```
Comment (dbetka) [inline client/src/store/team.ts:10]: default team color is needed here? #447744

**Review by dbetka — CHANGES_REQUESTED** [review]
@Bedpaw ideas/comments worth to note for the future:
1. Let admin see which team has which color.
2. Let team members and leader see what color they have.
3. Maybe it's worth to have info which team collected point in point popup above map (only for admin?)

Few little comments about current code and functionality is in review comments

File: client/src/components/pages/join-event.vue (hunk 1 above)
Comment (Bedpaw) [reply client/src/components/pages/join-event.vue]: done

File: client/src/store/team.ts (hunk 2 above)
Comment (Bedpaw) [reply client/src/store/team.ts]: removed

**Review by dbetka — APPROVED** [review]
Remember to note these comments and ideas on confluence: https://github.com/dbetka/harcmap/pull/620#pullrequestreview-960905870

Approve

**Pull request merged**